xbmc/guilib/GUIIncludes.cpp

```diff
@@ -18,72 +18,43 @@
+    "sliderwidth", "spinheight", "spinposx",      "spinposy",   "spinwidth",   "textoffsetx",
+    "textoffsety", "textwidth",  "timeperimage",  "top",        "width",
+};
+static_assert(std::ranges::is_sorted(CONSTANT_NODES));
+
+static constexpr std::string_view EXPRESSION_ATTRIBUTE = "condition";
+
+static constexpr std::array<std::string_view, 4> EXPRESSION_NODES = {
+    "enable",
+    "selected",
+    "usealttexture",
+    "visible",
+};
+static_assert(std::ranges::is_sorted(EXPRESSION_NODES));
+
+static bool Contains(auto&& container, std::string_view item)
 {
-  m_constantAttributes.insert("x");
-  m_constantAttributes.insert("y");
-  m_constantAttributes.insert("width");
-  m_constantAttributes.insert("height");
-  m_constantAttributes.insert("center");
-  m_constantAttributes.insert("max");
-  m_constantAttributes.insert("min");
-  m_constantAttributes.insert("w");
-  m_constantAttributes.insert("h");
-  m_constantAttributes.insert("time");
-  m_constantAttributes.insert("acceleration");
-  m_constantAttributes.insert("delay");
-  m_constantAttributes.insert("start");
-  m_constantAttributes.insert("end");
-  m_constantAttributes.insert("center");
-  m_constantAttributes.insert("border");
-  m_constantAttributes.insert("repeat");
-
-  m_constantNodes.insert("posx");
-  m_constantNodes.insert("posy");
-  m_constantNodes.insert("left");
-  m_constantNodes.insert("centerleft");
-  m_constantNodes.insert("right");
-  m_constantNodes.insert("centerright");
-  m_constantNodes.insert("top");
-  m_constantNodes.insert("centertop");
-  m_constantNodes.insert("bottom");
-  m_constantNodes.insert("centerbottom");
-  m_constantNodes.insert("width");
-  m_constantNodes.insert("height");
-  m_constantNodes.insert("offsetx");
-  m_constantNodes.insert("offsety");
-  m_constantNodes.insert("textoffsetx");
-  m_constantNodes.insert("textoffsety");
-  m_constantNodes.insert("textwidth");
-  m_constantNodes.insert("spinposx");
-  m_constantNodes.insert("spinposy");
-  m_constantNodes.insert("spinwidth");
-  m_constantNodes.insert("spinheight");
-  m_constantNodes.insert("radioposx");
-  m_constantNodes.insert("radioposy");
-  m_constantNodes.insert("radiowidth");
-  m_constantNodes.insert("radioheight");
-  m_constantNodes.insert("sliderwidth");
-  m_constantNodes.insert("sliderheight");
-  m_constantNodes.insert("itemgap");
-  m_constantNodes.insert("bordersize");
-  m_constantNodes.insert("timeperimage");
-  m_constantNodes.insert("fadetime");
-  m_constantNodes.insert("pauseatend");
-  m_constantNodes.insert("depth");
-  m_constantNodes.insert("movement");
-  m_constantNodes.insert("focusposition");
-
-  m_expressionAttributes.insert("condition");
-
-  m_expressionNodes.insert("visible");
-  m_expressionNodes.insert("enable");
-  m_expressionNodes.insert("usealttexture");
-  m_expressionNodes.insert("selected");
+  return std::ranges::binary_search(container, item);
```

Comment:
Now that you changed the approach and the std lib function returns a boolean, maybe it's not necessary to have a function for this anymore - it will not be changed to std::ranges::contains later.